File: Rohit Singhal/Trees/RangeSumofBST.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int sum=0;
    
    public void dfs(TreeNode root,int low,int high)
    {
        if(root==null)
            return;
        if(root.val>=low && root.val<=high)
        {
            sum+=root.val;
            dfs(root.right,low,high);
            dfs(root.left,low,high);
        }
        else if(root.val<low)
        {
            dfs(root.right,low,high);
        }
        else if(root.val>high)
        {
            dfs(root.left,low,high);
        }
        return;
    }
    
    public int rangeSumBST(TreeNode root, int low, int high) {
        dfs(root,low,high);
        return sum;
        
    }
}

File: Aman/Tree/Range Sum of BST.cpp
class Solution {
public:
    int res=0;
    int rangeSumBST(TreeNode* root, int l, int h) {
        if(root==NULL)
            return 0;
        rangeSumBST(root->left, l, h);
        rangeSumBST(root->right, l, h);
        if(root->val>=l && root->val<=h)
          res+=root->val;
        return res;
    }
};
